metadata_backend/api/handlers.py

```diff
@@ -1107,19 +1108,19 @@ def setup_static(self) -> Path:
 
 
 # Private functions shared between handlers
-async def _extract_xml_upload(req: Request, extract_one: bool = False) -> List[Tuple[str, str]]:
-    """Extract submitted xml-file(s) from multi-part request.
-
-    Files are sorted to spesific order by their schema priorities (e.g.
-    submission should be processed before study).
+async def _multipart_content(
+    req: Request, extract_one: bool = False, is_xml: bool = False
+) -> Tuple[List[Tuple[Any, str]], str]:
+    """Return list of each body part in a multipart request and determine whether request deals with xml or csv files.
 
     :param req: POST request containing "multipart/form-data" upload
-    :raises: HTTPBadRequest if request is not valid for multipart or multiple files sent. HTTPNotFound if
-    schema was not found.
-    :returns: content and schema type for each uploaded file, sorted by schema
-    type.
+    :param extract_one: boolean stating whether multiple files should be handled
+    :param is_xml: boolean stating if file can be expected to be XML
+    :raises: HTTPBadRequest for multiple different reasons
+    :returns: Either list of xml files or csv files for further parsing
```

Comment:
to my understanding it returns content not a list of files, did i understand it correctly?

Reply:
Yes. Technically it could be list of contents if multiple XML files are extracted at once but these comments were outdated from a previous commit and forgot to edit.